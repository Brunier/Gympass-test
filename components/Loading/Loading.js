import React from 'react';
import styled from "styled-components";

const Container = styled.div`
    text-align: center;
    display: ${props => props.display ? props.display : "block"};
    vertical-align: middle;
    position: relative;
    left: ${props => props.left ? props.left : "0"};
    top: 5px;
  
  img {
    width: ${props => props.width + "px"};
  }
 `;

export class Loading extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container {...this.props}>
                <img src={this.props.img ? this.props.img : "/static/loading.gif"} />
            </Container>
        )
    }
}

export default Loading;
