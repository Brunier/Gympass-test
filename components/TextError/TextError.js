import styled from 'styled-components';
import React from 'react';


let TextError = styled.span`
  color: red;
  font-size: 20px;
  margin-top: 20px;
  display: block;
`

export default TextError;

