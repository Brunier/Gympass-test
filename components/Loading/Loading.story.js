import React from 'react';
import { storiesOf } from '@storybook/react';
import Loading from './Loading';
import img from "../../static/loading.gif"

storiesOf('Loading', module)
    .add('Pequeno (20px)', () => (
        <Loading width={20} img={img} />
    ))
    .add('Medio (60px)', () => (
        <Loading width={60} img={img} />
    ))
    .add('Grande (100px)', () => (
        <Loading width={100} img={img} />
    ))
