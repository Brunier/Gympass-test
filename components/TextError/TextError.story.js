import React from 'react';
import { storiesOf } from '@storybook/react';
import TextError from './TextError';


storiesOf('TextError', module)
    .add('Normal', () => (
        <TextError>Texto Error</TextError>
    ))
