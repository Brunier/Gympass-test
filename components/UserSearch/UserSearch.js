import React from 'react'
import styled from "styled-components";
import TextError from '../TextError/TextError';
import Loading from '../Loading/Loading';
import Button from '../Button/Button'

let Container = styled.div`
  margin-top: 1%;
  text-align: center;
  color: #585656;
  
  input {
    padding: 10px 10px 10px 0;
    width: 15%;
    height: 27px;
    border: none;
    font-size: 23px;
    background-color: #fff;
    border: 2px solid #00757E;
    border-left: none;
    color: #585656;
  }
  
  label {
    background-color: white;    
    padding: 12px 7px 12px 12px;
    width: 110px;
    height: 23px;
    display: inline-block;
    border: none;
    font-size: 23px;
    border: 2px solid #00757E;
    border-right: none;
  }
`


class UserSearch extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            userName: null
        }
    }

    onFormSubmit = (e) =>  {
        e.preventDefault();
        this.props.setUser(this.state.userName);
    }

    render() {
        return(
            <Container>
                <form onSubmit={this.onFormSubmit}>
                    <div>
                        <label>Github.com/</label>
                        <input autoFocus placeholder={"Gympass"} onChange={event => this.setState({userName: event.target.value})}/>
                        {
                            !this.props.loading ?
                                <Button type="submit" text={"Procurar RepositÃ³rios"} disabled={!this.state.userName} />
                                : <Loading width={30} left={"2%"} display={"inline"} img={this.props.imgLoading}/>

                        }
                    </div>
                    {this.props.error ? <TextError>{this.props.error}</TextError> : ""}
                </form>
               </Container>

        )
    }
}


export default UserSearch;

