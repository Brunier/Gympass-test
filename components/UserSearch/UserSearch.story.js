import React from 'react';
import { storiesOf } from '@storybook/react';
import UserSearch from './UserSearch';
import img from '../../static/loading.gif'


storiesOf('UserRepos', module)
    .add('Normal', () => (
        <UserSearch  />
    ))

    .add('Loading', () => (
        <UserSearch loading={true} imgLoading={img}/>
    ))

    .add('Error', () => (
        <UserSearch error={"Aconteceu um erro :("}/>
    ))
