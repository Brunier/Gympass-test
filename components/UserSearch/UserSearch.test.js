import React from 'react';
import { shallow } from 'enzyme';
import TextError from '../TextError/TextError';
import Loading from '../Loading/Loading';
import renderer from 'react-test-renderer'

// Components
import UserSearch from './UserSearch';

function setup(props) {
    const wrapper = shallow(<UserSearch {...props} />);
    return { wrapper };
}

describe('UserSearch Test Suite', () => {
    it('Com nome vazio, o botão tem que estar desabilitado', () => {
        const { wrapper } = setup();
        wrapper.setState({ userName: '' });
        //expect(wrapper.find('button').exists()).toBe(true);
        expect(wrapper.find('button').prop("disabled")).toBe(true);
    });
    it('Com nome, o botão tem que estar habilitado', () => {
        const { wrapper } = setup();
        wrapper.setState({ userName: 'gympass' });
        //expect(wrapper.find('button').exists()).toBe(true);
        expect(wrapper.find('button').prop("disabled")).toBe(false);
    });
    it('Mostrar mensagem de erro', () => {
        const { wrapper } = setup({error: true});
        expect(wrapper.find(TextError).exists()).toBe(true);
    });
    it('Não mostrar a mensagem de erro', () => {
        const { wrapper } = setup({error: false});
        expect(wrapper.find(TextError).exists()).toBe(false);
    });
    it('Loading', () => {
        const { wrapper } = setup({loading: true});
        expect(wrapper.find(Loading).exists()).toBe(true);
    });
    it('Sem Loading', () => {
        const { wrapper } = setup({loading: false});
        expect(wrapper.find(Loading).exists()).toBe(false);
    });
    it('Snapshot', () => {
        const tree = renderer.create(<UserSearch />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
