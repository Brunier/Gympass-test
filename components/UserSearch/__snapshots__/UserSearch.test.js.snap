// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearch Test Suite Snapshot 1`] = `
<div
  className="sc-htpNat fpJEAM"
>
  <div>
    <label>
      Github.com/
    </label>
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Gympass"
    />
    <button
      disabled={true}
      onClick={[Function]}
    >
      Procurar Reposit√≥rios
    </button>
  </div>
  
</div>
`;
